src/app/routes/broker-ui/catalog-page/catalog-page/catalog-page.component.ts

```diff
@@ -143,4 +144,8 @@ export class CatalogPageComponent implements OnInit, OnDestroy {
       this.expandedFilterId = filterId;
     }
   }
+
+  onViewSelectionChange(viewMode: string) {
+    this.viewMode = viewMode;
+  }
```

Comment:
this could also be removed if accessing it directly from the template, if we don't do anything with it and basically just rely on the CD + the field for component state: `(stateChange)="viewMode = $event"`